Bottom: 3f6c457a7ad1117e6bc5b55a737bd1d17fa024c9
Top:    05beeff07df521fbc3f29dea7ffda4b6895cc25a
Author: Ian Kent <raven@themaw.net>
Date:   2013-06-28 13:22:06 +0800

autofs-5.0.7 - fix add null check in parse_server_string()

In a recent patch that added an error check for the case there's no ':' server
name delimiter parse_server_string() braces were not added which lead to an
incorrect parse error.


---

diff --git a/modules/lookup_ldap.c b/modules/lookup_ldap.c
index 35ea6ea..a2bfafd 100644
--- a/modules/lookup_ldap.c
+++ b/modules/lookup_ldap.c
@@ -1210,14 +1210,14 @@ static int parse_server_string(unsigned logopt, const char *url, struct lookup_c
 		const char *q = NULL;
 
 		/* Isolate the server. Include the port spec */
-		if (*ptr != '[')
+		if (*ptr != '[') {
 			q = strchr(ptr, ':');
 			if (!q) {
 				crit(logopt, MODPREFIX
 				     "LDAP server name not found in %s", ptr);
 				return 0;
 			}
-		else {
+		} else {
 			q = ++ptr;
 			while (*q == ':' || isxdigit(*q))
 				q++;
